torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -0,0 +1,67 @@
+"""
+Utils for caching the outputs of AOTAutograd
+"""
+from __future__ import annotations
+
+import logging
+
+import torch
+from torch._inductor.codecache import _ident, FxGraphCachePickler
+
+from .schemas import AOTConfig  # noqa: F401
+
+log = logging.getLogger(__name__)
+
+
+class AOTAutogradCacheDetails:
+    """
+    Object to capture all the details for a dynamo graph module relevant to computing
+    a safe and stable cache key for AOTAutograd.
+    """
+
+    def __init__(self, gm: torch.fx.GraphModule, config: AOTConfig):
+        self.gm = gm  # TODO: we'll handle different parts of the graph module
+        # TODO: We'll want to handle the full_args passed in as well
+        self.config = config  # Gets reduced by the Pickler
+
+    def debug_str(self) -> str:
+        return AOTAutogradCachePickler.debug_str(self)
+
+
+def _reduce_aot_config(config: AOTConfig):
+    """
+    Reduce the config to a stable key for caching.
```

Comment:
Hmmm - if we care about non-inductor backends working with the cache, then we probably want to include `fw_compiler`/`bw_compiler`/`decomposition_table` in some form here, right?

That is kind of annoyingly redundant for the inductor case - we are already caching on the inductor source code, so caching on the exact e.g. decomposition_table arguments is redundant. But two use cases are:

(1) if someone who is not inductor is using AOTAutograd as their backend, we probably want to recompile if they change the decompositions they pass in

(2) There are two nice backends that we have for debugging: `torch.compile(backend="aot_eager")` vs. `torch.compile(backend="aot_eager_decomp_partition")` that are mostly the same exact that they use different decomposition tables and partitioning functions. We probably want the cache key to encode their differences

Reply:
Yup, I think we do need to add some combination of these, though computing a hash key from the callables directly seems hard. Will add in a followup PR (there's also probably other global settings like `vmap` we'll need to add too)